scripts/setup-rg/rg-template.json

```diff
@@ -0,0 +1,376 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "projectPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Prefix used for naming resources."
+            },
+            "defaultValue": "mlos"
+        },
+        "vmSKU": {
+            "type": "string",
+            "defaultValue": "Standard_D2s_v3",
+            "metadata": {
+                "description": "VM SKU"
+            }
+        },
+        "vmAdminUsername": {
+            "type": "string"
+        },
+        "sshPublicKey": {
+            "type": "string"
+        },
+        "fileShareName": {
+            "type": "string",
+            "defaultValue": "mlos-file-share"
+        },
+        "resultsDbName": {
+            "type": "string"
+        },
+        "resultsDbAdminUsername": {
+            "type": "string",
+            "defaultValue": "mlos"
+        },
+        "resultsDbAdminPassword": {
+            "type": "securestring"
+        }
+    },
+    "functions": [],
+    "variables": {
+        "storageAccountName": "[toLower(concat(replace(parameters('projectPrefix'), '-', ''), 'storage'))]",
+        "vmName": "[concat(parameters('projectPrefix'), '-vm')]",
+        "networkInterfaceName": "[concat(variables('vmName'), '-ni-', uniqueString(resourceGroup().id))]",
+        "vnetName": "[concat(variables('vmName'), '-vnet')]",
+        "publicIpName": "[concat(variables('vmName'), '-ip')]",
+        "nsgName": "[concat(variables('vmName'), '-nsg')]",
+        "kvName": "[concat(parameters('projectPrefix'), '-kv')]"
+    },
+    "resources": [
+        {
+        "name": "[variables('publicIpName')]",
+        "type": "Microsoft.Network/publicIPAddresses",
+        "apiVersion": "2020-11-01",
+        "location": "[resourceGroup().location]",
+        "tags": {
+            "displayName": "PublicIPAddress"
+        },
+        "properties": {
+            "publicIPAllocationMethod": "Static",
+            "dnsSettings": {
+                "domainNameLabel": "[variables('vmName')]"
+            }
+        }
+    },
+    {
+        "name": "[variables('nsgName')]",
+        "type": "Microsoft.Network/networkSecurityGroups",
+        "apiVersion": "2020-11-01",
+        "location": "[resourceGroup().location]",
+        "properties": {
+            "securityRules": [
+                {
+                    "name": "CorpSSH",
+                    "properties": {
+                        "description": "Corp. IP SSH access",
+                        "protocol": "Tcp",
+                        "sourcePortRange": "*",
+                        "destinationPortRange": "22",
+                        "sourceAddressPrefix": "131.107.0.0/16",
```

Comment:
These IPs should not be exposed externally.  Let's keep that inside some `env.sh` files that can be sourced locally to help parameterize which IPs to by default allow connections from.
Ideally, automatically adding the caller IP would be very nice too.

Reply:
Parameterized all these specific IPs into the ARM parameter files!